Automattic/woocommerce-payments
# Added e2e test: Merchant / Subscriptions / Purchase subscription without signup fee

Fixes https://github.com/Automattic/woocommerce-payments/issues/1491

#### Changes proposed in this Pull Request

This PR introduces a new e2e test around the `Merchant / Subscriptions / Purchase subscription without signup fee` critical flow. Additionally, this adds in a price check step to the `Merchant / Subscriptions / Purchase subscription with signup fee` test.

#### Testing instructions

*Prereqs*

As these changes rely on having the WooCommerce Subscriptions plugin installed in the test environment to work, please make sure you have the following set in your `local.env` in the `tests/e2e/config` folder:

```
E2E_GH_TOKEN='githubPersonalAccessToken'
WC_SUBSCRIPTIONS_REPO='{owner}/{repo}'
```

To create the `E2E_GH_TOKEN`, follow [these instructions to generate a GitHub Personal Access Token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) and assign the `repo` scope to it.

Once that is configured, proceed to spin up the test environment and run through the test steps below.

* Make sure the CI passes
* Run the new tests locally either in headless or non-headless mode and verify they pass:
    * `npm run test:e2e ./tests/e2e/specs/subscriptions/purchase-no-sign-up-fee-subscription.spec.js`
    * `npm run test:e2e-dev ./tests/e2e/specs/subscriptions/purchase-no-sign-up-fee-subscription.spec.js`
* Run the changes to the existing tests locally in headless or non-headless mode and verify they pass:
    * `npm run test:e2e ./tests/e2e/specs/subscriptions/purchase-sign-up-fee-subscription.spec.js`
    * `npm run test:e2e-dev ./tests/e2e/specs/subscriptions/purchase-sign-up-fee-subscription.spec.js`
